### DISCUSSION TITLE
Get StatusCode/Quality of nodes from subscription

### DISCUSSION THREAD

**Discussion by Steineklopfer** [FreeOpcUa/opcua-asyncio#807] [Q&A] [answered]
So basically I have a Kepware Server that is running on my computer and it collects data from other servers like Siemens PLCs. 

Our program will get the data from this Kepware Server to use it to trigger measurements and other stuff.

We are currently not connecting directly to the Siemens PLCs because I think it will be problematic with all the certificates and stuff. So far it worked quite well with the Kepware's drivers. We might look into connecting directly later on once we get a better hang on  OPC UA and what is required do make successful connections with PLCs directly. 

Anyway my question is. I want to detect connection disruptions. Basically the PLC can go offline because of maintenance or power failure amongst other things. This is a construction environment where these things can happen quite often.

Provided that not all the power went off, our system might still be working but the Kepware Server will show item quality: Bad which is probably the "StatusCode" related to that node.

Now is it possible to conveniently read that along with the value while subscribing to several nodes?

**Comment by AndreasHeine** [comment]
in a SubscriptionHandler:
https://github.com/FreeOpcUa/opcua-asyncio/blob/033427b6a5bdf1a4a512e4976a410b6cc6291406/asyncua/common/subscription.py#L15
https://github.com/FreeOpcUa/opcua-asyncio/blob/033427b6a5bdf1a4a512e4976a410b6cc6291406/asyncua/common/subscription.py#L136
event_data -> class DataChangeNotif
https://github.com/FreeOpcUa/opcua-asyncio/blob/033427b6a5bdf1a4a512e4976a410b6cc6291406/asyncua/common/subscription.py#L53

-> "data.monitored_item.Value" should be an DataValue inkl. StatusCode 
https://github.com/FreeOpcUa/opcua-asyncio/blob/033427b6a5bdf1a4a512e4976a410b6cc6291406/asyncua/ua/uatypes.py#L921
and the Attribute which was subscribed is in data.subscription_data.attribute (data.subscription_data = class SubscriptionItemData)!

**Comment by Steineklopfer** [comment, marked as answer]
So basically this is what I have to do in the SubscriptionHandler?

```
    def datachange_notification(self, node, val, data):
        data = ua.DataValue(data)
        check = data.StatusCode_.value
        print(check)
```

**Reply by AndreasHeine** [reply]
data.monitored_item.Value.StatusCode_ <- Class StatusCode
https://github.com/FreeOpcUa/opcua-asyncio/blob/033427b6a5bdf1a4a512e4976a410b6cc6291406/asyncua/ua/uatypes.py#L295

**Reply by Steineklopfer** [reply]
Ok right :) When I add a .value at the end it works well.
Was a bit confused because my IDE didn't show the members as autocomplete.
Thanks!
